src/client/LiveStream.js

```diff
@@ -1419,97 +1412,122 @@ exports.StreamConfig = async function({name, customSettings={}}) {
  */
-exports.StreamDeactivate = async function({name}) {
+exports.StreamListUrls = async function({siteId}={}) {
   try {
-    let conf = await this.LoadConf({name});
-
-    let {objectId} = conf;
-    let libraryId = await this.ContentObjectLibraryId({objectId});
-
-    let mainMeta = await this.ContentObjectMetadata({
-      libraryId,
-      objectId
-    });
-    let status = await this.StreamStatus({name});
+    const STATUS_MAP = {
+      UNCONFIGURED: "unconfigured",
+      UNINITIALIZED: "uninitialized",
+      INACTIVE: "inactive",
+      STOPPED: "stopped",
+      STARTING: "starting",
+      RUNNING: "running",
+      STALLED: "stalled",
+    };
 
-    if(!mainMeta.live_recording) {
-      return {
-        state: status.state,
-        error: "Stream must be configured before deactivating"
-      };
-    }
+    if(!siteId) {
+      const tenantContractId = await this.userProfileClient.TenantContractId();
 
-    // Return error if the LRO is running
-    if(status.state !== "stopped") {
-      return {
-        state: status.state,
-        error: "Stream must be stopped before deactivating"
-      };
-    }
+      if(!tenantContractId) {
+        throw Error("No tenant contract ID configured");
+      }
 
-    let fabURI = mainMeta.live_recording.fabric_config.ingress_node_api;
-    // Support both hostname and URL ingress_node_api
-    if(!fabURI.startsWith("http")) {
-      // Assume https
-      fabURI = "https://" + fabURI;
+      siteId = await this.ContentObjectMetadata({
+        libraryId: tenantContractId.replace("iten", "ilib"),
+        objectId: tenantContractId.replace("iten", "iq__"),
+        metadataSubtree: "public/sites/live_streams",
+      });
     }
 
-    this.SetNodes({fabricURIs: [fabURI]});
-
-    let edgeWriteToken = mainMeta.live_recording.fabric_config.edge_write_token;
-
-    if(edgeWriteToken === undefined || edgeWriteToken === "") {
-      return {
-        state: "inactive",
-        error: "stream is already inactive"
-      };
-    }
-    let edgeMeta = await this.ContentObjectMetadata({
-      libraryId,
-      objectId,
-      writeToken: edgeWriteToken
+    const streamMetadata = await this.ContentObjectMetadata({
+      libraryId: await this.ContentObjectLibraryId({objectId: siteId}),
+      objectId: siteId,
+      metadataSubtree: "public/asset_metadata/live_streams",
+      resolveLinks: true,
+      resolveIgnoreErrors: true
     });
 
-    // Set stop time
-    edgeMeta.recording_stop_time = Math.floor(new Date().getTime() / 1000);
-    const newState = "inactive";
+    const activeUrlMap = {};
+    await this.utils.LimitedMap(
+      10,
+      Object.keys(streamMetadata || {}),
+      async slug => {
+        const stream = streamMetadata[slug];
+        let versionHash;
+
+        if(
+          stream &&
+          stream.sources &&
+          stream.sources.default &&
+          stream.sources.default["."] &&
+          stream.sources.default["."].container ||
```

Comment:
probably cleaner to use optional chaining for this

`stream?.sources?.default?.["."]?.container`

Reply:
I'm hesitant to use optional chaining in client-js because I don't think all apps that use it support it. I think there are a few Webpack 4 projects

Reply:
Oh, right this is the client. Fair point, no optional chaining